examples/python/model-chat.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import onnxruntime_genai as og
+import argparse
+import time
+
+def main(args):
+    if args.verbose: print("Loading model...")
+    if args.timings:
+        started_timestamp = 0
+        first_token_timestamp = 0
+
+    config = og.Config(args.model_path)
+    config.clear_providers()
+    if args.execution_provider != "cpu":
+        if args.verbose: print(f"Setting model to {args.execution_provider}")
+        config.append_provider(args.execution_provider)
+    model = og.Model(config)
+
+    if args.verbose: print("Model loaded")
+    
+    tokenizer = og.Tokenizer(model)
+    tokenizer_stream = tokenizer.create_stream()
+    if args.verbose: print("Tokenizer created")
+    if args.verbose: print()
+
+    search_options = {name:getattr(args, name) for name in ['do_sample', 'max_length', 'min_length', 'top_p', 'top_k', 'temperature', 'repetition_penalty'] if name in args}
+    search_options['batch_size'] = 1
+
+    if args.verbose: print(search_options)
+    
+    if args.chat_template:
+        if args.chat_template.count('{') != 1 or args.chat_template.count('}') != 1:
+            print("Error, chat template must have exactly one pair of curly braces with input word in it, e.g. '<|user|>\n{input} <|end|>\n<|assistant|>'")
+            exit(1)
+    else:
+        if model.type.startswith("phi"):
+            args.chat_template = '<|user|>\n{input} <|end|>\n<|assistant|>'
+        elif model.type.startswith("llama"):
+            args.chat_template = '<|start_header_id|>user<|end_header_id|>{input}<|eot_id|><|start_header_id|>assistant<|end_header_id|>'
+        else:
+            print("Chat Template is unknown for model type:", model.type, "and it can result in erroneous results, please specify --chat_template flag for better output, e.g. '<|user|>\n{input} <|end|>\n<|assistant|>'")
+            exit(1)
```

Comment:
Any reason for using exit(1) vs. throwing an exception. I think model-qa.py also does that, but I think we should use exceptions where possible.

Reply:
So, my thoughts were to exit the program if chat_template is empty as it can give unexpected output. Are you recommending to still continue the example even if chat_template is empty and throw an exception something like "Chat Template is empty and it can get unexpected output"?

Reply:
Throw an exception like this:

```python
raise ValueError(f"Chat template for model type {model.type} is not known. Please provide the chat template.")
```